### PR TITLE
dev-libs/sway: version bump to 1.0_beta2

### PR DESCRIPTION
**Pull request by SpeedJack**
[sway 1.0_beta2 released](https://github.com/swaywm/sway/releases/tag/1.0-beta.2).
**Should be merged after: #10527** 

Changes:

- Removed `libxcb[xkb]` dep: no longer required.
- Removed `src_install`: calling `newpamd` is not longer required since `meson.build` now properly install the pam module (commit https://github.com/swaywm/sway/commit/55ed2736b5a8d69e69ebf6a77fffe684309fd4ac).
- Add ability to disable building of swayidle, swaymsg, swaynag (and avoid installing completions files and man pages of disabled commands).
- Add support for building swaylock without pam (feature added with commit https://github.com/swaywm/sway/commit/c9773491207d36d6f5e651adcb7a64c7a015bba3).

Signed-off-by: Niccolò Scatena <speedjack95@gmail.com>
Package-Manager: Portage-2.3.52, Repoman-2.3.12

### CODE REVIEW

File: dev-libs/sway/sway-1.0_beta2.ebuild [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		inherit git-r3
+else
+		# Version format: major.minor-beta.betanum
+		SWAY_PV="$(ver_cut 1-2)-$(ver_cut 3).$(ver_cut 4)"
+		SRC_URI="https://github.com/swaywm/sway/archive/${SWAY_PV}.tar.gz -> ${P}.tar.gz"
+		S="${WORKDIR}/sway-${SWAY_PV}"
+		KEYWORDS="~amd64 ~x86"
+fi
+
+inherit eutils fcaps meson pam
```
Comment (Zlogene) [inline dev-libs/sway/sway-1.0_beta2.ebuild]: what eutils and pam for?

File: dev-libs/sway/sway-1.0_beta2.ebuild [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+EAPI=7
+
+if [[ ${PV} == 9999 ]] ; then
+		EGIT_REPO_URI="https://github.com/swaywm/sway.git"
+		inherit git-r3
+else
+		# Version format: major.minor-beta.betanum
+		SWAY_PV="$(ver_cut 1-2)-$(ver_cut 3).$(ver_cut 4)"
+		SRC_URI="https://github.com/swaywm/sway/archive/${SWAY_PV}.tar.gz -> ${P}.tar.gz"
```
Comment (Zlogene) [inline dev-libs/sway/sway-1.0_beta2.ebuild]: `${PN}`

File: dev-libs/sway/sway-1.0_beta2.ebuild [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		-e "/completions\/[a-z]\+\/_\?swaymsg/d" -i meson.build || die
+	use swaynag || sed -e "s/subdir('swaynag')//g" -e "/swaynag.[0-9].scd/d" \
+		-e "/completions\/[a-z]\+\/_\?swaynag/d" -i meson.build || die
+}
+
+src_configure() {
+	local emesonargs=(
+		"-Dsway-version=${SWAY_PV}"
+		$(meson_use wallpapers default-wallpaper)
+		$(meson_use zsh-completion zsh-completions)
```
Comment (Zlogene) [inline dev-libs/sway/sway-1.0_beta2.ebuild]: you do not need to double option and flag name if it matches

File: dev-libs/sway/sway-1.0_beta2.ebuild [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		$(meson_use fish-completion fish-completions)
```
Comment (Zlogene) [inline dev-libs/sway/sway-1.0_beta2.ebuild]: here too

File: dev-libs/sway/sway-1.0_beta2.ebuild [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	systemd? ( >=sys-apps/systemd-237 )
+	tray? ( >=sys-apps/dbus-1.10 )
+	X? ( x11-libs/libxcb:0= )"
+DEPEND="${RDEPEND}"
+BDEPEND="app-text/scdoc
+	virtual/pkgconfig"
+
+FILECAPS=( cap_sys_admin usr/bin/sway )
+
+src_prepare() {
```
Comment (Zlogene) [inline dev-libs/sway/sway-1.0_beta2.ebuild]: what is wrong with these options so you try to remove them conditionally?

File: dev-libs/sway/sway-1.0_beta2.ebuild [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+	>=dev-libs/wayland-protocols-1.14
+	x11-libs/cairo
+	x11-libs/libxkbcommon
+	x11-libs/pango
+	x11-libs/pixman
+	elogind? ( >=sys-auth/elogind-237 )
+	swaybar? ( x11-libs/gdk-pixbuf:2[jpeg] )
+	swaybg? ( x11-libs/gdk-pixbuf:2[jpeg] )
+	swaylock? (
+		pam? ( virtual/pam )
```
Comment (Zlogene) [inline dev-libs/sway/sway-1.0_beta2.ebuild]: will this really get fbsd keywords to make sense of virtual dependency?

File: dev-libs/sway/sway-1.0_beta2.ebuild (hunk 3 above)
Comment (SpeedJack) [reply dev-libs/sway/sway-1.0_beta2.ebuild]: They don't match. There is a final `s` in the option name for meson, while the flag name (which is global) doesn't have the `s`.

File: dev-libs/sway/sway-1.0_beta2.ebuild (hunk 4 above)
Comment (SpeedJack) [reply dev-libs/sway/sway-1.0_beta2.ebuild]: Same as above.

File: dev-libs/sway/sway-1.0_beta2.ebuild (hunk 2 above)
Comment (SpeedJack) [reply dev-libs/sway/sway-1.0_beta2.ebuild]: The name should contain also the package version, not only the package name (`${P}=${PN}-${PV}`). This is the approach used in all ebuilds that need to rename the archive. Try with `grep -r '.tar.gz -> ${P}.tar.gz' /usr/portage/`.

File: dev-libs/sway/sway-1.0_beta2.ebuild (hunk 1 above)
Comment (SpeedJack) [reply dev-libs/sway/sway-1.0_beta2.ebuild]: `eutils` is needed for `optfeature` (in `pkg_postinst`). `pam` is no longer needed, I'll remove it.

File: dev-libs/sway/sway-1.0_beta2.ebuild (hunk 5 above)
Comment (SpeedJack) [reply dev-libs/sway/sway-1.0_beta2.ebuild]: Version >=1.0 of Sway does not allow to disable building of optional binaries through compiler options, as it was in previous versions (eg. 0.15.2). These binaries are not required for Sway, and the user may want to substitute some of them with something else (a different status bar or screen locker).
So, to disable them, the ebuild now needs to remove them from the `meson.build` script.
Also, if the user does not want to install an optional binary, even the related man page and completions files should not be installed.

File: dev-libs/sway/sway-1.0_beta2.ebuild (hunk 6 above)
Comment (SpeedJack) [reply dev-libs/sway/sway-1.0_beta2.ebuild]: Sorry, I don't understand what `virtual/pam` has to do with fbsd.

**Pull request closed without merging**